modules/core/src/main/java/org/apache/ignite/internal/processors/rest/handlers/log/GridLogCommandHandler.java

```diff
@@ -0,0 +1,158 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ignite.internal.processors.rest.handlers.log;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.internal.GridKernalContext;
+import org.apache.ignite.internal.IgniteInternalFuture;
+import org.apache.ignite.internal.processors.rest.GridRestCommand;
+import org.apache.ignite.internal.processors.rest.GridRestResponse;
+import org.apache.ignite.internal.processors.rest.handlers.GridRestCommandHandlerAdapter;
+import org.apache.ignite.internal.processors.rest.request.GridRestLogRequest;
+import org.apache.ignite.internal.processors.rest.request.GridRestRequest;
+import org.apache.ignite.internal.util.future.GridFinishedFuture;
+import org.apache.ignite.internal.util.typedef.internal.U;
+
+import static org.apache.ignite.internal.processors.rest.GridRestCommand.LOG;
+
+/**
+ * Handler for {@link org.apache.ignite.internal.processors.rest.GridRestCommand#LOG} command.
+ */
+public class GridLogCommandHandler extends GridRestCommandHandlerAdapter {
+    /**
+     * Supported commands.
+    */
+    private static final Collection<GridRestCommand> SUPPORTED_COMMANDS = U.sealList(LOG);
+
+    /**
+     * Default log file name *
+     */
+    private static final String DEFAULT_LOG_PATH = "work/log/ignite.log";
+
+    /**
+     * Default log file start line number *
+     */
+    private static final int DEFAULT_FROM = 0;
+
+    /**
+     * Default log file end line number*
+     */
+    private static final int DEFAULT_TO = 1;
+
+    /**
+     * @param ctx Context.
+     */
+    public GridLogCommandHandler(GridKernalContext ctx) {
+        super(ctx);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Collection<GridRestCommand> supportedCommands() {
+        return SUPPORTED_COMMANDS;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public IgniteInternalFuture<GridRestResponse> handleAsync(GridRestRequest req) {
+        assert req != null;
+        if (req.command() == LOG) {
+            if (log.isDebugEnabled())
+                log.debug("Handling log REST request: " + req);
+            GridRestLogRequest req0 = (GridRestLogRequest) req;
+            try {
+                validateRange(req0);
+            } catch (IgniteCheckedException e){
+                return new GridFinishedFuture<>(new GridRestResponse(GridRestResponse.STATUS_FAILED, e.getMessage()));
+            }
+            int from = DEFAULT_FROM;
```

Comment:
Saikat,

Lets follow the Rest API docs [1] and implement support of 'from' and 'to' parameters the following way:
- if 'from' is set but 'to' is not then return an error;
- if 'to' is set but 'from' is not then return an error as well;
- if neither 'from' nor 'to' is set then return the first log line.
- if both parameters are set then just check that 'from' is less than 'to' and use them as is

[1] https://apacheignite.readme.io/docs/rest-api#section-request-parameters